### PR TITLE
*: support deleting user in v3 auth

### PR DESCRIPTION
**Pull request by mitake**
This commit adds a functionality of user deletion. It can be invoked
with the new user delete command.

Example usage:
$ ETCDCTL_API=3 etcdctl user delete usr1

### CODE REVIEW

File: auth/store.go [hunk 1]
```diff
@@ -103,6 +107,26 @@ func (as *authStore) UserAdd(r *pb.AuthUserAddRequest) (*pb.AuthUserAddResponse,
 	return &pb.AuthUserAddResponse{}, nil
 }
 
+func (as *authStore) UserDelete(r *pb.AuthUserDeleteRequest) (*pb.AuthUserDeleteResponse, error) {
+	plog.Noticef("deleting a user: %s", r.Name)
```
Comment (xiang90) [inline auth/store.go]: this is going to be confusing. say we logged we are deleting a user and then there is a not found error happened, the logging is strange... it seems like the deleting never finished. probably just remove the deleting log. the deleted log at line 124 should be good enough.

File: auth/store.go [hunk 2]
```diff
@@ -103,6 +107,26 @@ func (as *authStore) UserAdd(r *pb.AuthUserAddRequest) (*pb.AuthUserAddResponse,
...
+	if len(vs) != 1 {
+		return &pb.AuthUserDeleteResponse{}, ErrUserNotFound
+	}
+
+	tx.UnsafeDelete(authUsersBucketName, []byte(r.Name))
+
+	plog.Noticef("deleted a user: %s", r.Name)
+
+	return &pb.AuthUserDeleteResponse{}, nil
+
```
Comment (xiang90) [inline auth/store.go]: remove extra empty space?

File: etcdctl/ctlv3/command/user_command.go [hunk 3]
```diff
@@ -89,3 +98,15 @@ func userAddCommandFunc(cmd *cobra.Command, args []string) {
...
+
+// userDeleteCommandFunc executes the "user delete" command.
+func userDeleteCommandFunc(cmd *cobra.Command, args []string) {
+	if len(args) != 1 {
+		ExitWithError(ExitBadArgs, fmt.Errorf("user delete command requires user name as its argument."))
+	}
+
+	_, err := mustClientFromCmd(cmd).Auth.UserDelete(context.TODO(), args[0])
+	if err != nil {
+		ExitWithError(ExitError, err)
```
Comment (xiang90) [inline etcdctl/ctlv3/command/user_command.go]: probably we should print something out if delete succeeds?

**Comment by xiang90** [comment]
LGTM after fixing the comments.

File: auth/store.go (hunk 1 above)
Comment (mitake) [inline auth/store.go]: OK

File: auth/store.go (hunk 2 above)
Comment (mitake) [inline auth/store.go]: Oops, thanks for catching!

File: etcdctl/ctlv3/command/user_command.go (hunk 3 above)
Comment (mitake) [inline etcdctl/ctlv3/command/user_command.go]: etcdctl v2 doesn't print message in a case of successful deletion. So I followed the manner. Should I print something in v3? I don't have a strong opinion about this.

Comment (xiang90) [inline etcdctl/ctlv3/command/user_command.go]: me either... ok, let's keep the previous behavior then.

Comment (mitake) [inline etcdctl/ctlv3/command/user_command.go]: OK

**Comment by mitake** [comment]
@xiang90 updated for the comment, PTAL.

**Comment by xiang90** [comment]
Will merge when CI turns green. Thanks!

**Comment by mitake** [comment]
@xiang90 sorry... I misunderstood about the behavior of etcdctl v2. It prints a message in a case of successful user deletion :( Should I add the message? It seems that user add also prints a message in v2.

**Comment by xiang90** [comment]
@mitake No worries. We can add it.

**Comment by mitake** [comment]
@xiang90 thanks, I'll create a PR for the messages later

**Pull request merged**